src/Compilers/Core/Portable/Compilation/IExpression.cs

```diff
@@ -4,25 +4,26 @@
 
 namespace Microsoft.CodeAnalysis.Semantics
 {
-    /// <summary>
-    /// Represents a C# or VB expression.
-    /// </summary>
-    public interface IExpression : IOperation
+    public interface IHasArgumentsExpression : IOperation
     {
         /// <summary>
-        /// Result type of the expression.
+        /// Arguments of the invocation, excluding the instance argument. Arguments are in parameter order,
+        /// and params/ParamArray arguments have been collected into arrays. Default values are supplied for
+        /// optional arguments missing in source.
         /// </summary>
-        ITypeSymbol ResultType { get; }
+        ImmutableArray<IArgument> ArgumentsInParameterOrder { get; }
         /// <summary>
-        /// If the expression evaluates to a constant value, <see cref="Optional{Object}.HasValue"/> is true and <see cref="Optional{Object}.Value"/> is the value of the expression, and otherwise <see cref="Optional{Object}.HasValue"/> is false.
+        /// Find the argument supplied for a given parameter of the target method.
         /// </summary>
-        Optional<object> ConstantValue { get; }
+        /// <param name="parameter">Parameter of the target method.</param>
+        /// <returns>Argument corresponding to the parameter.</returns>
+        IArgument GetArgumentMatchingParameter(IParameterSymbol parameter);
     }
 
     /// <summary>
     /// Represents a C# or VB method invocation.
     /// </summary>
-    public interface IInvocationExpression : IExpression
+    public interface IInvocationExpression : IHasArgumentsExpression
```

Comment:
As of now, I think raising event is also covered by this. Right? If we don't separate it now, introducing a `RaiseEvent` operation might become a breaking change in the future.